### DISCUSSION TITLE
Easier replication management

### DISCUSSION THREAD

**Discussion by sandstrom** [dragonflydb/dragonfly#1212] [Ideas]
I've followed Dragonfly for a while, great progress!

A few small wishes for the coming year or so:

Instead of using commands to configure replication, we'd like to configure replication via some type of discovery mechanism or ring. Ideally with support (or compatibility with) Consul. 

I think Loki is a fairly good example here: https://grafana.com/docs/loki/latest/configuration/#ingester

That way, spinning up a new machine won't require any additional steps, it will automatically discover others and join.

**Comment by romange** [comment]
1. @sandstrom have you checked k8s dragonfly-operator? 

2. You can use the usual "replicaof" command to set master/slave status,

**Reply by sandstrom** [reply]
We're not using k8s, but I'll have a look.

Yes, I know. And after thinking about it a bit more, maybe it makes more sense to not have replication state hard-coded into a config file. So I'll take back my suggestion around having that configurable.
